### DISCUSSION TITLE
get_magnet_height

### DISCUSSION THREAD

**Discussion by SebGue** [Eomys/pyleecan#306] [Q&A]
Hello @BonneelP ,

with c483b58fa12f98eab6d17b32140fb2e0d46ce2d4 you removed the get_magnet_height methods for HoleMx.
Can you recall if this was intentional?

Best regards, Sebastian

**Comment by BonneelP** [comment]
Hello Sebastian,

Yes I removed it because it was not possible to generalize it for HoleUD since the definition was not clear. It implicitly assumed that all magnets are rectangular and that "height" makes sense. Let's also imagine a Hole with several magnets with different height, what should be returned ? I think that one day we will have a comp_height_active method for Hole that would work the same way as the one for Slot works (which would be a step forward using Holes as closed slot): The distance between two circles that contains the active part. 

For instance in pyleecan/Functions/FEMM/comp_FEMM_dict.py, I now use comp_height instead of get_magnet_height.
Do you need this method back and if yes what should be its definition ? 

Best regards,
Pierre

**Reply by SebGue** [reply]
Hello Pierre,

I use this methods in my internals. I haven't synced this repo so I didn't recognized the removal. So I can meanwhile also reintroduce the methods as internal.
For all actual Hole the  magnets should be rectangular, I think. The point is, that there is no method to get the 'active' surfaces height of the hole. But there is a comp_surface_magnet method that excatly do this thing regarding the magnets surface. 
If you say there will be Hole with different magnets, all these magnets need to have their own height. That would also be okay for me (in the future).

Best regards,
Sebastian

Edit: Moved OffTopic to Winding issue.
